We learned that some characters in the Cantonese SmolSent dataset, are exclusive to Cantonese.

So we became interested in how frequent this is since we suspect it has an impact on translation accuracy when a Mandarin model is used.

We found an online dictionary https://www.cantonese.sheik.co.uk/ that has descriptions for many Chinese characters. 

For some of these characters they are annotated with `This character is used in Cantonese, not Mandarin/Standard written Chinese.`.

We therefore made a web scraping script that looks for this sentence on the pages of all characters in our dataset.

In [4]:
import os
from character_analysis import load_character_type_data, fetch_data_for_chars, save_character_type_data, save_path, print_summary
if os.path.exists(save_path):
    character_type_dict = load_character_type_data()
else:
    character_type_dict = fetch_data_for_chars(all_chars) # This takes about 30 mins to terminate
    save_character_type_data(character_type_dict)
    
print_summary(character_type_dict)

Cantonese-only characters: 30
Not Cantonese-only characters: 2082
Unknown characters: 158
Cantonese-only characters: {'揸', '嗰', '咗', '鈪', '嘢', '焗', '嚟', '攞', '㩒', '揼', '睇', '啱', '唔', '煲', '㗎', '𦧲', '咁', '佢', '咩', '啫', '冧', '冇', '喺', '脷', '踭', '啲', '黐', '嘅', '抌', '哋'}
Sentences with at least one Cantonese-only character: 788 out of 863


The description is that only 30 of the 2270 different characters used are exclusive to Cantonese.

And 158 of them are not described in the online dictionary (Many of these inlcude English characters that are in the Cantonese sentences).

It is however worth to note that despite a small percentage of the characters being, Canto-only, they are used in 788 out of the 863 sentence. 